nginx-proxy/imgpath.js

```diff
@@ -0,0 +1,7 @@
+function file_path(r) {
+    var url = decodeURI(r.url)
```

Comment:
this isn't necessary, you can use `r.uri` below

(actually this code produces an `undefined` and therefore the hash is invalid)

Reply:
There is an `unquote()` in [file_storage code](https://github.com/thumbor/thumbor/blob/master/thumbor/result_storages/file_storage.py#L97) so if you remove decodeURI the resulting hash might differ from the python's one.

Reply:
Thanks! Updated.

Reply:
I think there's still a subtle issue if the url itself contains a remote resource fetched with `http://`, e.g. `http://localhost/cXs9D-PEju7UrVDeiMCOHF5P1nY=/fit-in/800x800/filters:fill(FFFFFF,true)/https://i.imgur.com/Nfn80ck.png` (produced with secret `abc`). It looks like `https://` within the uri gets normalized by nginx to `https:/` (only one slash, instead of two), so the hash value is different

The `file_path` that gets produced is `1e/db/6074c6ec438b78cd7433f71139b964068997` (hash value is `1edb6074c6ec438b78cd7433f71139b964068997`)

```python
>>> hashlib.sha1(unquote('/cXs9D-PEju7UrVDeiMCOHF5P1nY=/fit-in/800x800/filters:fill(FFFFFF,true)/https://i.imgur.com/Nfn80ck.png')).hexdigest()
'ef8d9f9e2ad6dbd1aea603c7843e7565a2b16e41'
>>> hashlib.sha1(unquote('/cXs9D-PEju7UrVDeiMCOHF5P1nY=/fit-in/800x800/filters:fill(FFFFFF,true)/https:/i.imgur.com/Nfn80ck.png')).hexdigest()
'1edb6074c6ec438b78cd7433f71139b964068997'
```

Reply:
Nginx changes url only in `r.url`, right? Inside python upstream you'll get this as normal, with two slashes?

If so, we could add some kind of regexp check for `http:/`/`https:/` inside njs script.

Reply:
Yes, it looks like `r.uri` changes. I'm not sure what other transformations nginx might apply, but we should probably be aware of those, or find a way to avoid those...